### ISSUE TITLE
`cypress/angular-signals` support for `InputSignalWithTransform`

### ISSUE THREAD

**Issue by bohoffi** [cypress-io/cypress#30682]
### What would you like?

#29264 brought support for the assignment of pure `InputSignal` which works fine as long as you do not plan on working with `InputSignalWithTransform` like:
```typescript
public readonly enablePreview: InputSignalWithTransform<boolean, unknown> = input(false, {
  transform: booleanAttribute
});
```

### Why is this needed?

The `componentProperties` of `MountConfig<T>` from [`cypress/angular-signals`](https://github.com/cypress-io/cypress/blob/5c21c4b40186670572acef9452ebd0875c296dc9/npm/angular-signals/src/mount.ts#L76) needs to be updated from
```typescript
componentProperties?: Partial<{
        [P in keyof T]: T[P] extends InputSignal<infer V> ? InputSignal<V> | WritableSignal<V> | V : T[P];
    }>;
```
to
```typescript
componentProperties?: Partial<{
        [P in keyof T]: T[P] extends InputSignal<infer V> | InputSignalWithTransform<infer V, unknown> ? InputSignal<V> | WritableSignal<V> | V : T[P];
    }>;
```

I'd be willing to provide a PR adding the missing type.

**Comment by AtofStryker** [comment]
@bohoffi thank you for opening this issue. I think `InputSignalWithTransform` was included in Angular 17 and we should be able to add this to the Cypress 14 release if that is the case. We will be merging the `cypress/angular-signals` harness upstream into `cypress/angular` for Cypress 14.

**Comment by bohoffi** [comment]
@AtofStryker yes, `InputSignalWithTransform` was introduced with Angular [17.2.0](https://github.com/angular/angular/releases/tag/17.2.0)  ([PR 54053](https://github.com/angular/angular/pull/54053)).

Thank you for taking it into account for Cypress 14 🙏 
Having it will resolve our `as unknown as` workaround :)

**Comment by AtofStryker** [comment]
@bohoffi I'm taking a look into this today. It looks like supporting the type is trivial, but the `transform` on the input signal doesn't actually get called mainly because of how we handle signals in the test harness via `Inferred Generic Type` or `Writable Signals` https://docs.cypress.io/app/component-testing/angular/examples#Testing-Signals. 

I can go ahead and fix the type, but it looks like its very hard to access the `transform`s and `alias` declared on the input signal since it lives in an internal symbol

**Comment by bohoffi** [comment]
@AtofStryker I'll see if I can find some time to look into it myself to find something to work with.

**Comment by vmasek** [comment]
Hello, is the current state still that CCT do not works with signal input transform?
From what I understood the `transform` function is never called, is that still a case?

What is proffered workaround, would a transform be called if I tested with a component that wraps the actual component I want to test and binds the input from template?

**Comment by logitimate** [comment]
Is there any update on this? We are doing a POC on playwright vs cypress and this would be a nice feature to have!

**Comment by AtofStryker** [comment]
@logitimate nothing as of yet. AFAIK Playwright doesn't support Angular component testing first class. We would be open to a contributor PR if someone is able to fix this!
